rust-runtime/aws-smithy-runtime/src/client/retries/strategy/standard.rs

```diff
@@ -45,6 +54,36 @@ impl StandardRetryStrategy {
         self.initial_backoff = initial_backoff;
         self
     }
+
+    pub fn with_max_backoff(mut self, max_backoff: Duration) -> Self {
+        self.max_backoff = max_backoff;
+        self
+    }
+
+    fn release_retry_permit(&self) -> ReleaseResult {
+        let mut retry_permit = self.retry_permit.lock().unwrap();
+        match retry_permit.take() {
+            Some(p) => {
+                drop(p);
+                APermitWasReleased
+            }
+            None => NoPermitWasReleased,
+        }
+    }
+
+    fn set_retry_permit(&self, new_retry_permit: OwnedSemaphorePermit) {
+        let mut old_retry_permit = self.retry_permit.lock().unwrap();
+        if let Some(p) = old_retry_permit.replace(new_retry_permit) {
+            // Whenever we set a new retry permit and it replaces the old one, we need to "forget"
+            // the old permit, removing it from the bucket forever.
+            p.forget()
```

Comment:
Curious to better understand `p.forget()` here. What would happen if we didn't execute `p.forget()`?

Reply:
Then the token would be returned to the bucket once dropped.